daemon/daemon.go

```diff
@@ -575,6 +576,14 @@ func (d *Daemon) Stop() error {
 		}
 		// wait for reboot to happen
 		logger.Noticef("Waiting for system reboot")
+		if sigCh != nil {
```

Comment:
I'd probably use something like:
```
select {
case <-timer.After(rebootWaitTimeout):
    return fmt.Error("reboot did not happen")
case <-sigCh:
    logger.Notice("got termination signal while waiting for reboot")
    return nil
}
```
but the current code works too.

Reply:
Thanks, I like this! Trying to add it, I get: `invalid operation: <-sigCh (receive from send-only type chan<- os.Signal)` though. I will merge as is and look into it a bit later.